sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConsumer.cs

```diff
@@ -237,5 +348,270 @@ protected EventHubConsumer()
+            {
+                await ChannelSyncRoot.WaitAsync().ConfigureAwait(false);
+
+                try
+                {
+                    // If the channel was the last active channel and publishing is still marked as active, take
+                    // the necessary steps to stop publishing.
+
+                    if ((_isPublishingActive) && (ActiveChannels.Count == 0))
+                    {
+                        _channelPublishingTokenSource?.Cancel();
+
+                        if (_channelPublishingTask != null)
+                        {
+                            try
+                            {
+                                await _channelPublishingTask.ConfigureAwait(false);
+                            }
+                            catch (TaskCanceledException)
+                            {
+                               // This is an expected scenario; no action is needed.
+                            }
+                        }
+
+                        _channelPublishingTokenSource?.Dispose();
+                        _isPublishingActive = false;
+                    }
+                }
+                finally
+                {
+                    ChannelSyncRoot.Release();
+                }
+            }
+
+            // Attempt to finalize the channel, signaling that no more writes should be
+            // expected to occur.
+
+            unsubscribeChannel?.Writer.TryComplete();
+        }
+
+        /// <summary>
+        ///   Begins the background process responsible for receiving from the partition
+        ///   and publishing to all subscribed channels.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to signal the request to cancel the background publishing.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        private Task StartBackgroundChannelPublishingAsync(CancellationToken cancellationToken) =>
+            Task.Run(async () =>
+            {
+                var failedAttemptCount = 0;
+                var receivedItems = default(IEnumerable<EventData>);
+                var retryDelay = default(TimeSpan?);
+                var activeException = default(Exception);
+
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    try
+                    {
+                        // Receive items in batches and then write them to the subscribed channels.  The channels will naturally
+                        // block if they reach their maximum queue size, so there is no need to throttle publishing.
+
+                        receivedItems = await InnerConsumer.ReceiveAsync(BackgroundPublishReceiveBatchSize, BackgroundPublishingWaitTime, cancellationToken).ConfigureAwait(false);
+
+                        await Task.WhenAll(
```

Comment:
Might be worth unrolling this into foreach loops if we don't find a way to use a single channel.